### PR TITLE
Finish #33906 User sign up

### PR DESCRIPTION
**Pull request by trantube94**
I have created a pull request, check it please!

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
+class UsersController < ApplicationController
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.new user_params
+    if @user.save
+      flash[:success] = I18n.t("Welcome_framgia")
```
Comment (phamkykhoi) [inline app/controllers/users_controller.rb]: Remove ( )

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -7,6 +7,9 @@
   <%= csrf_meta_tags %>
 </head>
 <body>
+  <% flash.each do |message_type, message| %>
+    <div class="alert alert-<%= message_type %>"><%= message %></div>
+  <% end %>
 
 <%= yield %>
```
Comment (phamkykhoi) [inline app/views/layouts/application.html.erb]: Indent

File: app/views/shared/_error_messages.html.erb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+<% if @user.errors.any? %>
+  <div id="error_explanation">
+    <div class="alert alert-danger">
+      The form contains <%= pluralize(@user.errors.count, "error") %>.
```
Comment (phamkykhoi) [inline app/views/shared/_error_messages.html.erb]: Move "The form contains " to Settings

File: app/controllers/users_controller.rb [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+      render "new"
+    end
+  end
+
+  def show
+    @user = User.find params[:id]
+  end
+
+  private
+    def user_params
```
Comment (dieunb) [inline app/controllers/users_controller.rb]: indent

File: app/helpers/users_helper.rb [hunk 5]
```diff
@@ -0,0 +1,7 @@
+module UsersHelper
+  def gravatar_for user
+    gravatar_id = Digest::MD5::hexdigest user.email.downcase
+    gravatar_url = "https:// secure.gravatar.com/avatar/#{gravatar_id}"
+    image_tag(gravatar_url, alt: user.name, class: "gravatar")
```
Comment (dieunb) [inline app/helpers/users_helper.rb]: remove `()`

File: app/helpers/users_helper.rb [hunk 6]
```diff
@@ -0,0 +1,7 @@
+module UsersHelper
+  def gravatar_for user
```
Comment (dieunb) [inline app/helpers/users_helper.rb]: do you want to add `size`?

File: app/views/layouts/application.html.erb [hunk 7]
```diff
@@ -1,14 +1,20 @@
...
-  <%= stylesheet_link_tag    'application', media: 'all', 'data-turbolinks-track' => true %>
-  <%= javascript_include_tag 'application', 'data-turbolinks-track' => true %>
-  <%= csrf_meta_tags %>
-</head>
-<body>
 
-<%= yield %>
+  <head>
+    <title>Internship</title>
+    <%= stylesheet_link_tag    'application', media: 'all', 'data-turbolinks-track' => true %>
```
Comment (phamkykhoi) [inline app/views/layouts/application.html.erb]: ' ' => " "
not use => 
use :

File: app/views/layouts/application.html.erb [hunk 8]
```diff
@@ -1,14 +1,20 @@
...
+    <%= javascript_include_tag 'application', 'data-turbolinks-track' => true %>
```
Comment (phamkykhoi) [inline app/views/layouts/application.html.erb]: ' ' => " "
not use => 
use :

**Pull request merged**